src/corppa/poetry_detection/core.py

```diff
@@ -112,6 +135,38 @@ def to_dict(self) -> dict[str, Any]:
                     json_dict[key] = value
         return json_dict
 
+    def to_csv(self) -> dict[str, int | str]:
+        """
+        Returns a CSV-friendly dict of the poem excerpt. Unlike `to_dict`, unset optional
+        fields are included and set to empty strings.
+        """
+        csv_dict = {}
+        for key, value in asdict(self).items():
+            if value is None:
+                # Set unset fields as empty strings
+                csv_dict[key] = ""
+            elif type(value) is set:
+                # Convert sets to comma-separated lists
+                csv_dict[key] = ", ".join(value)
+            else:
+                csv_dict[key] = value
+        return csv_dict
+
+    def strip_whitespace(self) -> "Excerpt":
+        """
+        Returns a copy of the excerpt such that leading and trailing whitespace
+        have been removed from the text. In addition to the text, the start
+        and end indices will also be updated.
```

Comment:
Suggested edit to make it a bit more readable

```suggestion
        Return a copy of this excerpt with any leading and trailing whitespace
        removed from the text and start and end indices updated to match any changes.
```